File: Ruby/k01.rb
class K01
  def execute(num)
    case num
    when 1
      question1
    when 2
      question2
    when 3
      question3
    when 4
      question4
    else
      return
    end
  end

  def question1
    puts 'Hello World! ようこそRuby言語の世界へ！'
  end

  def question2
    puts 'Hello World!'
    puts 'ようこそ'
    puts 'Ruby言語の世界へ！'
  end

  def question3
    puts format('整数：%d', 12345)
    puts format('実数：%f', 123.456789)
    puts format('文字：%c', 'X')
    puts format('文字列：%s', 'ABCdef')
  end

  def question4
    puts '              ##'
    puts '             #  #'
    puts '             #  #'
    puts '            #    #'
    puts '           #      #'
    puts '         ##        ##'
    puts '       ##            ##'
    puts '    ###                ###'
    puts ' ###       ##    ##       ###'
    puts '##        #  #  #  #        ##'
    puts '##         ##    ##         ##'
    puts ' ##     #            #     ##'
    puts '  ###     ##########     ###'
    puts '     ###              ###'
    puts '        ##############'
  end
end

File: Go/K01.go
package main

import "fmt"

type K01 struct {}

func (k01 K01) question1() {
	fmt.Println("Hello World!　ようこそGo言語の世界へ！")
}

func (k01 K01) question2() {
	fmt.Println("Hello World!")
	fmt.Println("ようこそ")
	fmt.Println("Go言語の世界へ！")
}

func (k01 K01) question3() {
	fmt.Println(fmt.Sprintf("整数：%d", 12345))
	fmt.Println(fmt.Sprintf("実数：%f", 123.456789))
	fmt.Println(fmt.Sprintf("文字：%c", 'A'))
	fmt.Println(fmt.Sprintf("文字列：%s", "ABCdef"))
}

func (k01 K01) question4() {
	fmt.Println("              ##")
	fmt.Println("             #  #")
	fmt.Println("             #  #")
	fmt.Println("            #    #")
	fmt.Println("           #      #")
	fmt.Println("         ##        ##")
	fmt.Println("       ##            ##")
	fmt.Println("    ###                ###")
	fmt.Println(" ###       ##    ##       ###")
	fmt.Println("##        #  #  #  #        ##")
	fmt.Println("##         ##    ##         ##")
	fmt.Println(" ##     #            #     ##")
	fmt.Println("  ###     ##########     ###")
	fmt.Println("     ###              ###")
	fmt.Println("        ##############")
}

func (k01 K01) execute(num int) {
	switch num {
	case 1:
		k01.question1()
	case 2:
		k01.question2()
	case 3:
		k01.question3()
	case 4:
		k01.question4()
	default:
		break
	}
}